expr/render_test.go

```diff
@@ -0,0 +1,304 @@
+		{el: mDestination.Element, want: mSource.Element},
+	}
+	for i, tt := range tests {
+		t.Run(fmt.Sprint(i), func(t *testing.T) {
+			got := relatedSoftwareSystems(tt.el)
+			if got != nil {
+				if got[0].Name != tt.want.Name {
+					t.Errorf("Got %s, wanted %s", got[0].Name, tt.want.Name)
+				}
+			}
+		})
+	}
+
+}
+
+func Test_RelatedContainers(t *testing.T) {
+	mSource, mDestination := makeRelatedContainers(t)
+	tests := []struct {
+		el   *Element
+		want *Element
+	}{
+		{el: mSource.Element, want: mDestination.Element},
+		{el: mDestination.Element, want: mSource.Element},
+	}
+	for i, tt := range tests {
+		t.Run(fmt.Sprint(i), func(t *testing.T) {
+			got := relatedContainers(tt.el)
+			if got != nil {
+				if got[0].Name != tt.want.Name {
+					t.Errorf("Got %s, wanted %s", got[0].Name, tt.want.Name)
+				}
+			}
+		})
+	}
+
+}
+
+func Test_RelatedComponents(t *testing.T) {
+	mSource, mDestination := makeRelatedComponents(t)
+	tests := []struct {
+		el   *Element
+		want *Element
+	}{
+		{el: mSource.Element, want: mDestination.Element},
+		{el: mDestination.Element, want: mSource.Element},
+	}
+	for i, tt := range tests {
+		t.Run(fmt.Sprint(i), func(t *testing.T) {
+			got := relatedComponents(tt.el)
+			if got != nil {
+				if got[0].Name != tt.want.Name {
+					t.Errorf("Got %s, wanted %s", got[0].Name, tt.want.Name)
+				}
+			}
+		})
+	}
+
+}
+
+func Test_RelatedInfrastructureNodes(t *testing.T) {
+	mSource, mDestination := makeRelatedInfrastructureNodes(t)
+	tests := []struct {
+		el   *Element
+		want *Element
+	}{
+		{el: mSource.Element, want: mDestination.Element},
+		{el: mDestination.Element, want: mSource.Element},
+	}
+	for i, tt := range tests {
+		t.Run(fmt.Sprint(i), func(t *testing.T) {
+			got := relatedInfrastructureNodes(tt.el)
+			if got != nil {
+				if got[0].Name != tt.want.Name {
+					t.Errorf("Got %s, wanted %s", got[0].Name, tt.want.Name)
+				}
+			}
+		})
+	}
+
+}
+
+func Test_RelatedContainerInstances(t *testing.T) {
+	mSource, mDestination := makeRelatedContainerInstances(t)
+	tests := []struct {
+		el   *Element
+		want *Element
+	}{
+		{el: mSource.Element, want: mDestination.Element},
+		{el: mDestination.Element, want: mSource.Element},
+	}
+	for i, tt := range tests {
+		t.Run(fmt.Sprint(i), func(t *testing.T) {
+			got := relatedContainerInstances(tt.el)
+			if got != nil {
+				if got[0].Name != tt.want.Name {
+					t.Errorf("Got %s, wanted %s", got[0].Name, tt.want.Name)
+				}
+			}
+		})
+	}
+
+}
+
+func Test_AddInfluencers(t *testing.T) {
+	mModel, mSoftwareSystem := makeSimModel(t)
+
+	mContainerView, mRelationships := makeContainerView(*mSoftwareSystem, t)
+	mModel.Systems[0].Relationships = mRelationships
+	mSoftwareSystem.Relationships = mRelationships
+
+	tests := []struct {
+		el *ContainerView
+	}{
+		{el: &mContainerView},
+	}
+	for i, tt := range tests {
+		t.Run(fmt.Sprint(i), func(t *testing.T) {
+			addInfluencers(tt.el)
+			/*got := cv
```

Comment:
Should this test be removed? or finalized?

Reply:
It should be finalised - as per the comment above I'll try and do it this weekend..

Reply:
Great, please let me know when it's ready for final review, thank you!